compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -246,7 +246,7 @@ impl AdtDefData {
         if Some(did) == tcx.lang_items().owned_box() {
             flags |= AdtFlags::IS_BOX;
         }
-        if Some(did) == tcx.lang_items().manually_drop() {
+        if Some(did) == tcx.lang_items().manually_drop() || tcx.has_attr(did, sym::manually_drop) {
```

Comment:
```suggestion
        if Some(did) == tcx.has_attr(did, sym::manually_drop) {
```
feels weird to have both a lang item and an attribute